server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/DeletedMessagesVaultTest.java

```diff
@@ -410,6 +465,232 @@ public void vaultEndpointShouldRestoreMessageForSharer() {
             .body(ARGUMENTS + ".list.subject", hasItem(SUBJECT));
     }
 
+    @Category(BasicFeature.class)
+    @Test
+    public void vaultExportShouldSendMailsToShareeWhenJmapDeleteMessage() throws Exception {
+        bartSendMessageToHomer();
+        WAIT_TWO_MINUTES.until(() -> listMessageIdsForAccount(homerAccessToken).size() == 1);
+        String messageIdOfHomer = listMessageIdsForAccount(homerAccessToken).get(0);
+
+        homerDeletesMessages(listMessageIdsForAccount(homerAccessToken));
+        WAIT_TWO_MINUTES.until(() -> listMessageIdsForAccount(homerAccessToken).size() == 0);
+
+        exportVaultContent(ExportRequest
+            .userExportFrom(HOMER)
+            .sharee(BART)
+            .query(MATCH_ALL_QUERY));
+
+        WAIT_TWO_MINUTES.until(() -> listMessageIdsForAccount(bartAccessToken).size() == 2);
+        String exportingMessageId = listMessageIdsForAccount(bartAccessToken).get(0);
+        String fileLocation =
+            with()
```

Comment:
Extract this complex JMAP query behind an explicit method: `bartReceivesTheSharingTheExportMail`